packages/astro/src/compiler/transform/head.ts

```diff
@@ -115,6 +115,116 @@ export default function (opts: TransformOptions): Transformer {
               },
             ],
           },
+          {
+            start: 0,
+            end: 0,
+            type: 'Expression',
+            codeChunks: ['Astro.pageScripts.map(script => (', '))'],
+            children: [
+              {
+                start: 0,
+                end: 0,
+                type: 'Expression',
+                codeChunks: ['script.src ? (', ') : (', ')'],
+                children: [
+                  {
+                    type: 'Element',
+                    name: 'script',
+                    attributes: [
+                      {
+                        type: 'Attribute',
+                        name: 'type',
+                        value: [
+                          {
+                            type: 'Text',
+                            raw: 'module',
+                            data: 'module'
+                          }
+                        ]
+                      },
+                      {
+                        type: 'Attribute',
+                        name: 'src',
+                        value: [
+                          {
+                            start: 0,
+                            end: 0,
+                            type: 'MustacheTag',
+                            expression: {
+                              start: 0,
+                              end: 0,
+                              type: 'Expression',
+                              codeChunks: ['script.src'],
+                              children: [],
+                            },
+                          }
+                        ]
+                      },
+                      {
+                        type: 'Attribute',
+                        name: 'data-astro',
+                        value: [
+                          {
+                            type: 'Text',
+                            raw: 'hoist',
+                            data: 'hoist'
+                          }
+                        ]
+                      }
+                    ],
+                    start: 0,
+                    end: 0,
+                    children: [],
+                  },
+
+                  { // TODO change
```

Comment:
The `data-astro` attribute setup here looks good to me, is this TODO comment still needed?